extension/data.ts

```diff
@@ -296,93 +287,308 @@ class RcxDict {
+      (charCode >= 12448 && charCode <= 12543) ||
+      (charCode >= 65382 && charCode <= 65437)
+    );
+  }
 
-  cv: number[] = [
-    0x30f4, 0xff74, 0xff75, 0x304c, 0x304e, 0x3050, 0x3052, 0x3054, 0x3056,
-    0x3058, 0x305a, 0x305c, 0x305e, 0x3060, 0x3062, 0x3065, 0x3067, 0x3069,
-    0xff85, 0xff86, 0xff87, 0xff88, 0xff89, 0x3070, 0x3073, 0x3076, 0x3079,
-    0x307c,
-  ];
-  cs: number[] = [0x3071, 0x3074, 0x3077, 0x307a, 0x307d];
+  convertToHiragana(str: string): string {
+    let result = '';
+    for (let i = 0; i < str.length; i++) {
+      const char = str.charAt(i);
+      const nextChar = i + 1 <= str.length - 1 ? str.charAt(i + 1) : null;
+      const semiVoiced =
+        nextChar === this.hiraganaLookup['semiVoicedSoundMark'];
+      const voiced = nextChar === this.hiraganaLookup['voicedSoundMark'];
+      const key = semiVoiced
+        ? char + nextChar
+        : voiced
+        ? char + nextChar
+        : char;
+      const hiragana = this.hiraganaLookup[key];
+      result += hiragana !== undefined ? hiragana : char;
+
+      if (semiVoiced || voiced) {
+        i++;
+      }
+    }
+    return result;
+  }
 
   wordSearch(
     word: string,
     doNames: boolean,
     max?: number
   ): DictEntryData | null {
     let i;
-    let u;
-    let v;
-    let reason: string;
-    let p;
     const trueLen = [0];
     const entry = RcxDict.createDefaultDictEntry();
-
-    // half & full-width katakana to hiragana conversion
-    // note: katakana vu is never converted to hiragana
-
-    p = 0;
-    reason = '';
-    for (i = 0; i < word.length; ++i) {
-      u = v = word.charCodeAt(i);
-
-      // Skip Zero-width non-joiner used in Google Docs between every
-      // character.
-      if (u === 8204) {
-        p = 0;
-        continue;
-      }
-
-      if (u <= 0x3000) {
-        break;
-      }
-
-      // full-width katakana to hiragana
-      if (u >= 0x30a1 && u <= 0x30f3) {
-        u -= 0x60;
-      } else if (u >= 0xff66 && u <= 0xff9d) {
-        // half-width katakana to hiragana
-        u = this.ch[u - 0xff66];
-      } else if (u === 0xff9e) {
-        // voiced (used in half-width katakana) to hiragana
-        if (p >= 0xff73 && p <= 0xff8e) {
-          reason = reason.substr(0, reason.length - 1);
-          u = this.cv[p - 0xff73];
-        }
-      } else if (u === 0xff9f) {
-        // semi-voiced (used in half-width katakana) to hiragana
-        if (p >= 0xff8a && p <= 0xff8e) {
-          reason = reason.substr(0, reason.length - 1);
-          u = this.cs[p - 0xff8a];
-        }
-      } else if (u === 0xff5e) {
-        // ignore J~
-        p = 0;
-        continue;
-      }
-
-      reason += String.fromCharCode(u);
-      // need to keep real length because of the half-width semi/voiced
-      // conversion
-      trueLen[reason.length] = i + 1;
-      p = v;
+    let isKana = false;
+    for (let i = 0; i < word.length; i++) {
+      isKana = this.isKana(word[i].charCodeAt(0));
```

Comment:
This is a bit weird. Essentially, isKana will always be whatever the last character is.

Should it be kana if all characters or kana or if any character is kana?

In that case, something like `Array.from(word).all(this.isKana))` would probably be clearer. (Didn't test that code so might not be exact)

Reply:
Yeah close. Maybe the `.all` is python? Here is what I refactored it to. LMK what you think.

```isKana = Array.from(word).every((char) => this.isKana(char.charCodeAt(0)));```